lib/manager/bundler/extract.ts

```diff
@@ -38,33 +38,18 @@ export async function extractPackageFile(
     if (rubyMatch) {
       res.compatibility = { ruby: rubyMatch[1] };
     }
-    let gemMatch: RegExpMatchArray;
-    let gemDelimiter: string;
-    for (const delimiter of delimiters) {
-      const gemMatchRegex = `^gem ${delimiter}([^${delimiter}]+)${delimiter}(,\\s+${delimiter}([^${delimiter}]+)${delimiter}){0,2}`;
-      if (regEx(gemMatchRegex).test(line)) {
-        gemDelimiter = delimiter;
-        gemMatch = gemMatch || regEx(gemMatchRegex).exec(line);
-      }
-    }
+    const gemMatchRegex = /^\s*gem\s+(['"])(?<depName>[^'"]+)\1(\s*,\s*(?<currentValue>(['"])[^'"]+\5(\s*,\s*\5[^'"]+\5)?))?/;
```

Comment:
This will now allow unbalanced quotes, but would that even happen except for in an incorrect Gemfile?

Reply:
I see the gitlab example you called out. I guess that tradeoff is worth it.

Reply:
I would extract unbalanced/heterogeneous quotes, but not sure if versioning scheme can deal with it